python/ray/train/tests/test_base_trainer.py

```diff
@@ -236,10 +235,12 @@ def train_loop(self):
     tune.run(trainer.as_trainable(), num_samples=4)
 
 
+@patch("ray.available_resources", ray.cluster_resources)
```

Comment:
Is `ray.available_resources` deterministic?

Reply:
You mean `cluster_resources`? Yes, as long as we don't consider autoscaling (not an issue here). In this test, it will always return 4 CPUs as expected.

Reply:
Ah, yeah. I meant `cluster_resources`.

Is it not an issue because we don't perform autoscaling in CI?

Reply:
That's correct - and we also initialize ray with fixed 4 CPUs using the `ray_start_4_cpus` fixture. Therefore, for this test, `cluster_resources` will always return 4 CPUs.